lighthouse-core/gather/driver.js

```diff
@@ -885,7 +886,9 @@ class Driver {
     const monitorForInsecureState = this._monitorForInsecureState();
     const securityCheckPromise = monitorForInsecureState.promise.then(securityMessages => {
       return function() {
-        throw new LHError(LHError.errors.INSECURE_DOCUMENT_REQUEST, {securityMessages});
+        if (!ignoreHttpsErrors) {
```

Comment:
I'm not sure this is entirely what we want since we'll have already fallen into the wrong `Promise.race` branch below. I think we want a `_waitForever` version of `_waitForNothing`


```js
const securityCheckPromise = ignoreHttpsErrors ? this._waitForever() : <promise as it is now>

Reply:
makes sense, I'll refactor